### ISSUE TITLE
false positive invalid-name on attrs classvar

### ISSUE THREAD

**Issue by Redoubts** [pylint-dev/pylint#10525]
### Bug description

consider

```python
from typing import ClassVar

import attrs


class X:
    A: ClassVar[int] = 1

    def __init__(self) -> None:
        self.b: int = 2


@attrs.define
class Y:
    A: ClassVar[int] = 1
    b: int = 2
```

I notice that with default settings, `X.A` is fine but `Y.A` gets an `invalid-name` lint. I think `A` is supposed to match on `class-attribute-naming-style`, which is `any` by default, but this appears to match `attr-naming-style` or something like it

### Configuration

```ini

```

### Command used

```shell
pylint --rcfile /dev/null --disable all --enable invalid-name x.py
```

### Pylint output

```python
x.py:15:4: C0103: Attribute name "A" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

10/10 would lint again

### Pylint version

```shell
pylint 3.3.7
astroid 3.3.11
Python 3.11.13
```

### OS / Environment

Debian

### Additional dependencies

```python

```

**Comment by zenlyj** [comment]
Thanks for creating this issue and sharing your observations. You are correct,  the message is incorrectly triggered because `Y.A` is treated as an instance variable. This will require a patch in astroid.
